### DISCUSSION TITLE
From the "sub" parameter after get_user() how to get user name/email?

### DISCUSSION THREAD

**Discussion by thetorque** [rayluo/identity#6] [Q&A] [answered]
Do I need to use the "sub" parameter to query the username/email by other means?

**Comment by rayluo** [comment]
The user object which you obtained that `sub` from, should have already contain other keys, such as a "preferred_username", if not "email". Does that work?

**Comment by thetorque** [comment]
Sorry I should have been more clear.

I meant to get the email that's used to log in to Azure. I believe it's called User Principal Name.

**Comment by rayluo** [comment, marked as answer]
From this library's perspective, the user object is actually all the content inside the ID token. [Azure AD's ID token contains these info](https://learn.microsoft.com/en-us/azure/active-directory/develop/id-token-claims-reference#payload-claims). The "preferred_username" and/or "email" is approximately what you want. Note that the same doc specifically [calls out](https://learn.microsoft.com/en-us/azure/active-directory/develop/id-token-claims-reference#use-claims-to-reliably-identify-a-user) that UPN is not necessarily unique.

**Comment by thetorque** [comment]
Understood. Thank you very much for your response and I really appreciate the work you put into this package. It's a lifesaver for me.
